### PR TITLE
Added TypeScript types

### PR DESCRIPTION
**Pull request by EXtremeExploit**
ADDED: TypeScript for index.js
ADDED: Dependencies on README.md
ADDED: test/ to .gitignore

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
   "author": "Discord Bot List <support@discordbots.org> (https://discordbots.org)",
   "license": "Apache-2.0",
   "dependencies": {
+    "discord.js": "^11.3.0",
```
Comment (tonkku107) [inline package.json]: Remove discord.js from dependencies.
Not everyone using this module is using discord.js and installing discord.js just for this module is overkill

File: README.md [hunk 2]
```diff
@@ -7,6 +7,10 @@ A module for interacting with the discordbots.org API
 ## Documentation
 Documentation can be found [here](https://discordbots.org/api/docs#jslib)
 
+### Dependencies
```
Comment (tonkku107) [inline README.md]: The readme doesn't need a dependencies section.
npm handles those via package.json

**Review by tonkku107 — CHANGES_REQUESTED** [review]

**Comment by EXtremeExploit** [comment]
Check PR now, i commited your requests

File: index.js [hunk 3]
```diff
@@ -1,7 +1,14 @@
 const snekfetch = require('snekfetch');
 const API = 'https://discordbots.org/api/';
+const { Bot, Bots, BotsQuery, Stats, User, UserSocial, Votes } = require('./ts/index');
+const { Client, Shard } = require('discord.js')
```
Comment (tonkku107) [inline index.js]: You're still requiring discord.js, this would cause errors

File: ts/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
+import { Client, Shard } from "discord.js";
```
Comment (tonkku107) [inline ts/index.d.ts]: Same here

**Review by tonkku107 — CHANGES_REQUESTED** [review]

**Comment by EXtremeExploit** [comment]
Fixed

**Comment by tonkku107** [comment]
Would you be able to type client and shard in some other way? (Which is not specific to any lib)

**Comment by EXtremeExploit** [comment]
Fixed x2

**Comment by tonkku107** [comment]
I don't see a new commit?
If you meant that `any` type, that's not really proper typing, is it?

**Comment by EXtremeExploit** [comment]
then if you want a proper typing, i can set that `any` to Client, but that's specific to a library

**Comment by tonkku107** [comment]
#3

**Pull request closed without merging**